web/src/list_widget.ts

```diff
@@ -580,6 +581,20 @@ export function create<Key, Item = Key>(
                 widget.hard_redraw();
             }
         },
+
+        clear_text_filter() {
+            const $filter = opts.filter?.$element;
+            if (!$filter) {
+                return;
+            }
+            const $container = $filter.parent(".search-container");
+            $container.find(".clear-filter").on("click", (e) => {
```

Comment:
Corresponding `off` call for this handler should be added in `clear_event_handlers`.